src/backend/libpq/rangerrest.c

```diff
@@ -464,6 +466,11 @@ static int call_ranger_rest(CURL_HANDLE curl_handle, const char* request)
 		}
 		else
 		{
+			if (switchToMaster)
+			{
+				/* master's RPS has recovered, switch from standby's RPS to master's RPS */
+				elog(NOTICE, "switch from standby's RPS to master's RPS");
```

Comment:
Thanks! You are right.
If switch to master and failed again, then talk with standby success, this notice message is also printed, a condition should be added.